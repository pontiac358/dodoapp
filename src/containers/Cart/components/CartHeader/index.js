import React from 'react';
import PropTypes from 'prop-types';
import {Link } from 'react-router-dom'
import styled from '@emotion/styled'

const Container = styled.div`
    color: #fff;
`;
const CartLink = styled(Link)`
    color: #fff;
    display: inline-block;
    text-align: right;
    padding: 2px;
    text-decoration: underline;
`;
CartLink.displayName = 'CartLink'

const CartHeader = ({ cart:{count, totalPrice} }) => {

    return (
        <Container>
            <CartLink to={'/cart'}>
                кол: <strong>{ count }</strong><br/>
                сумма: <strong>{ totalPrice } р</strong>
            </CartLink>
        </Container>
    );
}

CartHeader.propTypes = {
    cart: PropTypes.object
}

export default CartHeader;