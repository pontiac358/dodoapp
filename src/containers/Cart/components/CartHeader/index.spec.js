import React from 'react'
import { shallow } from 'enzyme'
import CartHeader  from '../CartHeader'

describe('ProductsList component', () => {
    const props = {
        cart:{
            count: 1,
            totalPrice: 100
        }
    };

    const wrapper = shallow(<CartHeader {...props} />);

    it("render correctly CartHeader container", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders <CartLink /> wrapper', () => {
        expect(wrapper.find('CartLink')).toHaveLength(1)
    })
});

