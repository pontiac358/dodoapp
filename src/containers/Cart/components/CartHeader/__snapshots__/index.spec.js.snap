// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList component render correctly CartHeader container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartHeader
    cart={
      Object {
        "count": 1,
        "totalPrice": 100,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef(render)
        replace={false}
        to="/cart"
      >
        кол: 
        <strong>
          1
        </strong>
        <br />
        сумма: 
        <strong>
          100
           р
        </strong>
      </ForwardRef(render)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "кол: ",
          <strong>
            1
          </strong>,
          <br />,
          "сумма: ",
          <strong>
            100
             р
          </strong>,
        ],
        "replace": false,
        "to": "/cart",
      },
      "ref": null,
      "rendered": Array [
        "кол: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": "strong",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "сумма: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              100,
              " р",
            ],
          },
          "ref": null,
          "rendered": Array [
            100,
            " р",
          ],
          "type": "strong",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": [Function],
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "
    color: #fff;
    display: inline-block;
    text-align: right;
    padding: 2px;
    text-decoration: underline;
",
        ],
        "defaultProps": Object {
          "replace": false,
        },
        "displayName": "CartLink",
        "render": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "
    color: #fff;
",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef(render)
          replace={false}
          to="/cart"
        >
          кол: 
          <strong>
            1
          </strong>
          <br />
          сумма: 
          <strong>
            100
             р
          </strong>
        </ForwardRef(render)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "кол: ",
            <strong>
              1
            </strong>,
            <br />,
            "сумма: ",
            <strong>
              100
               р
            </strong>,
          ],
          "replace": false,
          "to": "/cart",
        },
        "ref": null,
        "rendered": Array [
          "кол: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "strong",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "сумма: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                100,
                " р",
              ],
            },
            "ref": null,
            "rendered": Array [
              100,
              " р",
            ],
            "type": "strong",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": [Function],
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    color: #fff;
    display: inline-block;
    text-align: right;
    padding: 2px;
    text-decoration: underline;
",
          ],
          "defaultProps": Object {
            "replace": false,
          },
          "displayName": "CartLink",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "
    color: #fff;
",
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
